### ISSUE TITLE
Application Only OAuth

### ISSUE THREAD

**Issue by prshreshtha** [not-an-aardvark/snoowrap#140]
I would like to have a user-less context with an installed app. Application Only OAuth looks like the best way to do that. The relevant `grant_type`s are here: https://github.com/reddit-archive/reddit/wiki/OAuth2#application-only-oauth

The following request:
```http
POST /api/v1/access_token HTTP/1.1
Host: www.reddit.com
Content-Type: application/x-www-form-urlencoded
Authorization: Basic <client_id>
Cache-Control: no-cache

grant_type=https%3A%2F%2Foauth.reddit.com%2Fgrants%2Finstalled_client&\
device_id=DO_NOT_TRACK_THIS_DEVICE
```
responds with
```json
{
    "access_token": "<access_token>",
    "token_type": "bearer",
    "device_id": "DO_NOT_TRACK_THIS_DEVICE",
    "expires_in": 3600,
    "scope": "*"
}
```
Is this currently possible with `snoowrap`? If not, do you foresee any difficulties/API breaking if this were implemented?

Thanks for creating and maintaining this awesome library! 😃

**Comment by not-an-aardvark** [comment]
Hi, thanks for creating an issue. This isn't currently possible with snoowrap, although I would accept a pull request to add it if you're interested in working on it.

I'm imagining it as a static method on `snoowrap` that would accept a client ID/device ID/user agent as parameters, and return a Promise for a snoowrap instance (similar to how the [`fromAuthCode`](https://not-an-aardvark.github.io/snoowrap/snoowrap.html#.fromAuthCode__anchor) method works).

**Comment by prshreshtha** [comment]
Sounds good. Although I think the changes will extend to `updateAccessToken` etc.

**Comment by napindc** [comment]
@not-an-aardvark  I have a nodejs server that wants to pull recent reddit posts from several subreddits, which doesn't need links to a user account at all. I think it is related to this issue, but when I do the following:

```
const r = new snoowrap({
        userAgent: userAgent,
        clientId: clientId,
        clientSecret: secret,
        refreshToken: refreshToken
    });
    r.getHot().map(post => post.title).then(console.log);

```

I get this error: ```Unhandled rejection StatusCodeError: 400 - {"message":"Bad Request","error":400}```

How come I can't just pass in my clientId / secret that I got form https://www.reddit.com/prefs/apps/, and start making calls?

**Comment by not-an-aardvark** [comment]
@napindc Thanks for your comment, but I don't think your problem is related to this issue. Can you please open a new issue?

**Comment by not-an-aardvark** [comment]
(sorry, hit the wrong button and temporarily closed this issue)

**Comment by SpyTec** [comment]
Fixed in #207
